### DISCUSSION TITLE
opening files on Linux (version 11.3)

### DISCUSSION THREAD

**Discussion by ousia** [orbitalquark/textadept#228] [Q&A] [answered]
Sorry for this basic question, but this makes Textadept really hard to use for me.

I’m afraid I can only open files in Textadept from the program itself (dropping a the file icon on an open buffer from Textadept).

If (not having Textadept already opened) I run `textadept a.txt`, all info from `~/.textadept/session` is gone in the next start.

The same happens when opening a file with an extension linked to be opened with Textadept (or when dropping the file icon on the program icon) .

Sorry for the basic question, but is there no way to prevent the opening of files rewrites the previous session info  (and that new opened files are added to the ones opened in last session)?
 
BTW, I guess this won’t be different in other OSes (but I haven’t checked it on _Windows_).

Many thanks for your help.

**Comment by paaguti** [comment]
Hi Pablo,
OK one thing is opening files and the other is the session... I think you
need to be a it more explicit.
What do you need? Carrying over the file list from a previous session?


On Thu, 7 Jul 2022 at 19:00, Pablo Rodríguez ***@***.***>
wrote:

> Sorry for this basic question, but this makes Textadept really hard to use
> for me.
>
> I’m afraid I can only open files in Textadept from the program itself
> (dropping a the file icon on an open buffer from Textadept).
>
> If (not having Textadept already opened) I run textadept a.txt, all info
> from ~/.textadept/session is gone in the next start.
>
> The same happens when opening a file with an extension linked to be opened
> with Textadept (or when dropping the file icon on the program icon) .
>
> Sorry for the basic question, but is there no way to prevent the opening
> of files rewrites the previous session info (and that new opened files are
> added to the ones opened in last session)?
>
> BTW, I guess this won’t be different in other OSes (but I haven’t checked
> it on *Windows*).
>
> Many thanks for your help.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/228>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4GRMYVDELSP2E6N4BDVS4EJFANCNFSM526DTQRA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Reply by ousia** [reply]
Sorry, @paaguti, the behavior looked so surprising to me, that I forgot to explain it properly.

Geany calls it “Load files from last session”. This is the basic idea: each time Textadept opens a file (no matter how), recover files from previous session.

This would also imply that the previous session info doesn’t get lost when opening a new file.

Many thanks for your help.

**Comment by mhwombat** [comment]
Perhaps [this function](https://github.com/orbitalquark/textadept/wiki/import-default-session) would help? I haven't tried it yet myself.

**Reply by ousia** [reply]
Many thanks for your reply, @mhwombat.

This is in the right direction. Many thanks again, because what it seemed to be a bug is really a feature.

That function would be perfect if it could check if the previous session was imported and import it automatically if it wasn’t (with the newly opened files).

Any suggestions on how to implement that?

**Reply by ousia** [reply]
Many thanks for your help, @mhwombat.

Sorry for unmarking your reply an the answer.

I’m afraid GitHub only allows one answer per discussion 😅.

**Comment by mhwombat** [comment]
At the moment I don't see an easy way to do that without a modification to Textadept itself.

I've been working my own Textadept module to do session management, because I want to have a persistent session for each directory. I'm using `textadept.session.save` and `textadept.session.load` for that purpose.

So I looked at how I might accomplish what you want. Unfortunately, `textadept.session.load` replaces any files you might already have loaded. What would be nice is if this function added a boolean parameter that you can use to keep any existing buffers.

**Reply by rgieseke** [reply]
Couldn't you just parse and load the files from the session file (or check whether they are open)?
Not sure it's easy to detect whether a single file has been opened, there is `events.APPLEEVENT_ODOC` for macOS but maybe it's enough to check after `events.INITIALIZED` whether the session files have been loaded?

Re-using and changing `session.load` might be tricky because it also recreates state (like position or open buffer) - what should that be if you open a new file, likely the new file ...

**Reply by ousia** [reply]
@mhwombat and @rgieseke, many thanks for your replies. I really appreciate Textadept and your help.

I can’t see why such a basic feature isn’t available (not to mention by default).

I cannot code, but how about checking whether the files from previous session have been loaded and if not reload them (plus the files just loaded when invoking the program)?

Sorry, but it’s really surprising that such basic functionality is so hard to get.

I wonder whether @orbitalquark might have something to comment here.

Many thanks for your help again.

**Comment by orbitalquark** [comment, marked as answer]
For some reason, GitHub is not accepting my e-mail replies. Sorry about that :( I had previously responded with this:

You can use the `-n` or `--nosession` command line argument. If you pair this with `-f` or `--force`, then you will always open a new instance of Textadept without opening or saving the default session.

See https://orbitalquark.github.io/textadept/manual.html#running for command line arguments.

On Windows, you can set up a shortcut to "textadept.exe" that passes command line arguments, and then invoke that for opening files like *a.txt*.

**Reply by gabdub** [reply]
I noticed that when you run `textadept.exe -s session file.c` , the last opened files are kept and the new one (file.c in this example) is also opened, but if you run `textadept.exe file.c -s session` only the last opened files are shown. The session overwrites all buffers, so the order is important.

It would be great if we could set in init.lua that we want the `-s session` option by default.

**Reply by ousia** [reply]
Many thanks for your reply, @orbitalquark.

As a proposal, it would be great to have the proper options to add session and to force a unique instance in `init.lua`.

Many thanks again for your help.

**Comment by rgieseke** [comment]
> I can’t see why such a basic feature isn’t available (not to mention by default).

I guess it's because many users keep Textadept open all the time so the session restoring is mainly used after a reboot.

**Reply by ousia** [reply]
Sorry, @rgieseke, I forgot your reply.

I don’t have my computer on long periods of time.

**Reply by rgieseke** [reply]
No worries, this was more meant as an explanation why it might be a useful feature but hasn't been requested in the past. I guess for many users the typical workflow is, start computer, open browser, open editor, keep them open ... but that doesn't work if you restart often and start the editor with opening a file.

**Reply by ousia** [reply]
I realize I’m also “abusing” the editor by simply using it to write.

I mean, simply typing data with no commands inside (if such a difference makes sense).

Not all the time, but I have files with nothing else than pure texts (again, pure data, no commands [considering commands also any code and/or tag]) inside.

And I don’t always have the editor open. I didn’t realize how uncommon this was.

**Comment by paaguti** [comment]
Hi


maybe you can use the "Recent files in the File Open dialog"... I do in
Linux and macOS

Best, /PA

On Fri, 8 Jul 2022 at 20:36, Pablo Rodríguez ***@***.***>
wrote:

> Sorry, @paaguti <https://github.com/paaguti>, the behavior looked so
> surprising to me, that I forgot to explain it properly.
>
> Geany calls it “Load files from last session”. This is the basic idea:
> each time Textadept opens a file (no matter how), recover files from
> previous session.
>
> This would also imply that the previous session info doesn’t get lost when
> opening a new file.
>
> Many thanks for your help.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/228#discussioncomment-3109783>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4HRCRH3CH7EQK2Q6PLVTBYKFANCNFSM526DTQRA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Comment by orbitalquark** [comment]
You can use the `-n` or `--nosession` command line argument. If you pair this with `-f` or `--force`, then you will always open a new instance of Textadept without opening or saving the default session.

See https://orbitalquark.github.io/textadept/manual.html#running for command line arguments.
